lib/schemes/oauth2.js

```diff
@@ -198,4 +202,85 @@ export default class Oauth2Scheme {
 
     return true // True means a redirect happened
   }
+
+  // ---------------------------------------------------------------
+  // Watch axios requests for token expiration
+  // Refresh tokens if token has expired
+  // ---------------------------------------------------------------
+
+  initializeTokenRefreshOnRequest () {
+    const { $axios } = this.$auth.ctx.app
+    let isRefreshing = false
+
+    $axios.onRequest(async config => {
+      let token = this.$auth.getToken(this.name)
+      let refreshToken = this.$auth.getRefreshToken(this.name)
+
+      // Token or "refresh token" does not exists
+      if (!token || !refreshToken || !token.length || !refreshToken.length) {
+        return config
```

Comment:
I _think_ a simple return (without config) also means to skip middleware

Reply:
I think you are right. It looks like returning a falsy statement is the same as an unchanged config!

https://github.com/nuxt-community/axios-module/blob/225903871048428e9413f43caa159dafb21a93cd/lib/plugin.js#L20

Reply:
I just checked, it did not work without returning the config...